sdk/core/src/error/mod.rs

```diff
@@ -0,0 +1,354 @@
+use std::borrow::Cow;
+use std::fmt::{Debug, Display};
+
+mod azure_core_errors;
+mod hyperium_http;
+mod macros;
+
+/// A convience alias for `Result` where the error type is hard coded to `Error`
+pub type Result<T, O = std::convert::Infallible> = std::result::Result<T, Error<O>>;
+
+/// The kind of error
+///
+/// The classification of error is intentionally fairly coarse.
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum ErrorKind<O> {
+    /// An error specific to a certain operation
+    Operation(O),
+    /// An HTTP status code that was not expected
+    UnexpectedOperation { status: u16 },
```

Comment:
Across our other language SDKs, this typically would include "HTTP" or "request", like `RequestFailed { status: u16 }`.